### PR TITLE
feat: Migrate and Enhance Spring Analyzer for Kotlin (#251)

### PR DESCRIPTION
**Pull request by ksg97031**
### Overview
I've made some significant updates to migrate our existing Java Spring analyzer to a Kotlin Spring analyzer. This includes adding new classes like `KotlinLexer` and `KotlinParser`, as well as various refactorings and improvements to enhance functionality.

### Key Changes
1. **Addition of KotlinLexer Class**
   - Introduced a new `KotlinLexer` class that inherits from `MiniLexer` to tokenize Kotlin code.
   - Improved the handling of string and character literals in the lexer.

2. **Addition and Enhancement of KotlinParser Class**
   - Added a new `KotlinParser` class specifically for analyzing Spring-Kotlin applications.
   - Refactored and enhanced the functions for parsing class fields and annotations.
   - Enhanced parsing capabilities to better handle Kotlin code.

3. **Porting Java Spring Analyzer to Kotlin Spring**
   - Ported the existing Java Spring analyzer to a Kotlin Spring analyzer to support Kotlin environments.
   - Refactored Spring detectors to work with both Kotlin and Java files.

### Review Requests
Please let me know if you find any bugs or if there are any areas that need improvement.

Thanks a lot!

### CODE REVIEW

**Review by hahwul — CHANGES_REQUESTED** [review]
Hiya @ksg97031 
Your code looks fantastic as always! I really appreciate your attention to detail :D

I noticed that you included the `gradle-wrapper.jar` file. Do you think it's essential for our testing flow? Just curious about your thoughts on this.

Other than that, everything seems perfect. If there are no further issues, I'll go ahead and approve and merge this PR.

Thanks!

**Comment by ksg97031** [comment]
> Hiya @ksg97031 Your code looks fantastic as always! I really appreciate your attention to detail :D
> 
> I noticed that you included the `gradle-wrapper.jar` file. Do you think it's essential for our testing flow? Just curious about your thoughts on this.
> 
> Other than that, everything seems perfect. If there are no further issues, I'll go ahead and approve and merge this PR.
> 
> Thanks!

File: spec/functional_test/fixtures/kotlin_spring/gradle/wrapper/gradle-wrapper.jar
Comment (ksg97031) [inline spec/functional_test/fixtures/kotlin_spring/gradle/wrapper/gradle-wrapper.jar:1]: The gradle-wrapper.jar is used to ensure that the same version of Gradle is used across all environments, but it is not strictly necessary for the project to function.

Comment (ksg97031) [reply spec/functional_test/fixtures/kotlin_spring/gradle/wrapper/gradle-wrapper.jar:1]: I've removed the gradle-wrapper.jar file as suggested.

**Review by hahwul — APPROVED** [review]

**Pull request merged**